src/main/scala/firrtl/passes/InferWidths.scala

```diff
@@ -10,358 +10,128 @@ import firrtl._
 import firrtl.ir._
 import firrtl.Utils._
 import firrtl.Mappers._
-
-object InferWidths extends Pass {
-  type ConstraintMap = collection.mutable.LinkedHashMap[String, Width]
-
-  def solve_constraints(l: Seq[WGeq]): ConstraintMap = {
-    def unique(ls: Seq[Width]) : Seq[Width] =
-      (ls map (new WrappedWidth(_))).distinct map (_.w)
-    // Combines constraints on the same VarWidth into the same constraint
-    def make_unique(ls: Seq[WGeq]): ListMap[String,Width] = {
-      ls.foldLeft(ListMap.empty[String, Width])((acc, wgeq) => wgeq.loc match {
-        case VarWidth(name) => acc.get(name) match {
-          case None => acc + (name -> wgeq.exp)
-          // Avoid constructing massive MaxWidth chains
-          case Some(MaxWidth(args)) => acc + (name -> MaxWidth(wgeq.exp +: args))
-          case Some(width) => acc + (name -> MaxWidth(Seq(wgeq.exp, width)))
+import firrtl.Implicits.{constraint2width, width2constraint}
+
+class InferWidths extends Pass {
```

Comment:
Quibble: Did you consider adding a object InferWidths with a `def apply: InferWidths = new InferWidths`, a lot of list of transforms would have remained unchanged

Reply:
I'll add it.